### DISCUSSION TITLE
Add Filterable for Select

### DISCUSSION THREAD

**Discussion by orgnizedmess** [charmbracelet/huh#559] [Ideas]
**Is your feature request related to a problem? Please describe.**
I was working on recreating the older version of `gum choose` (prior to using `huh`) with `huh`'s Select field. The older version didn't have an input field or filtering, but Select has filtering enabled without an input field. So typing the `/` key and typing input filters the list, but I don't see what I typed, which doesn't feel right (I also have the help section hidden). 

This also makes the current `gum choose` very similar in functionality to `gum filter` rather than it being a component with different functionality.

I looked for a way to disable filtering, and noticed that Select does not have the option to do so. However [MultiSelect does](https://github.com/charmbracelet/huh/blob/main/field_multiselect.go#L115), and I'd like to suggest adding the option to Select as well.

**Describe the solution you'd like**
Add the `filterable` value to the `Select` struct in [field_select.go](https://github.com/charmbracelet/huh/blob/main/field_select.go), and the corresponding `Filterable()` function to set the value of `filterable`.

**Describe alternatives you've considered**
Use the component as is, with filtering enabled (aka how `gum choose` currently works)

**Additional context**
None

**Comment by joaogmguimaraes** [comment]
> So typing the `/` key and typing input filters the list, but I don't see what I typed, which doesn't feel right (I also have the help section hidden).

If no title is added to the menu, the filter text will not be displayed (this also occurs with the `MultiSelect`).

**Comment by joaogmguimaraes** [comment]
I forgot to add the test cases.
First with title and second without

<figure>
    <img src="https://vhs.charm.sh/vhs-7Lqjxe0sT7Az8aPgFES4F8.gif" width=400 />
</figure>

<figure>
    <img src="https://vhs.charm.sh/vhs-5dR72fE420Ewe2z0yYjNGk.gif" width=400 />
</figure>

<img src="https://stuff.charm.sh/vhs/badge.svg">

**Comment by bashbunni** [comment]
Hey @pjg11, thank you so much for wanting to make this project better! I've converted this to a discussion so we can talk through it and keep our backlog clear. If it does end up being something we want to add, we'll promote it back to an issue when we are ready to work on it.

If we commit to this feature, that's a feature we will need to maintain for the lifespan of this project. So, since "Yes is forever", we are carefully considering new features.

I do think this makes sense to implement given that we already support this in `MultiSelect`. Side note: Since this issue was created, Gum has been decoupled from Gum.
